src/lib/cache/deep-research-cache.ts

```diff
@@ -47,10 +48,22 @@ export async function getDeepResearchResult(
     const data = await redisClient.get(key);
     if (!data) return null;
     
-    const result = JSON.parse(data) as DeepResearchResult;
+    // Parse the data and validate through schema to ensure correct structure
+    const parsedData = JSON.parse(data);
+    
     // Convert string timestamp back to Date object
-    result.timestamp = new Date(result.timestamp);
-    return result;
+    if (typeof parsedData.timestamp === 'string') {
+      parsedData.timestamp = new Date(parsedData.timestamp);
+    }
+    
+    // Fix any items that might have been serialized incorrectly
+    if (Array.isArray(parsedData.results)) {
```

Comment:
this should not be done manually, zod should take care of it, this is a stupid bandaid solution that doesn't fix the root cause. if the setup is good this isn't needed.